### PR TITLE
Reduce the complexity of linear_algebra/src/polynom_for_points.py

### PR DESCRIPTION
**Pull request by MaximSmolskiy**
### Describe your change:

Reduce the complexity of linear_algebra/src/polynom_for_points.py from 23 to 17

Fixes: #7942

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: linear_algebra/src/polynom_for_points.py [hunk 1]
```diff
@@ -24,96 +24,79 @@ def points_to_polynomial(coordinates: list[list[int]]) -> str:
...
+        return "The program cannot work out a fitting polynomial."
 
-        if len(coordinates) == 1 and coordinates[0][0] == 0:
-            check = 2
-            solved = "x=0"
-    except Exception:
-        check = 3
+    set_x = {x for x, _ in coordinates}
+    if len(set_x) == 1:
+        return "x=" + str(coordinates[0][0])
```
Comment (cclauss) [inline linear_algebra/src/polynom_for_points.py:35]: Shorter, faster, and easier to read.
```suggestion
        return f"x={coordinates[0][0]}"
```

File: linear_algebra/src/polynom_for_points.py [hunk 2]
```diff
@@ -24,96 +24,79 @@ def points_to_polynomial(coordinates: list[list[int]]) -> str:
...
-            count += 1
+    count = 0
+    solved = "f(x)="
 
-        return solved
+    while count < x:
+        remove_e: list[str] = solution[count].split("E")
+        if len(remove_e) > 1:
+            solution[count] = remove_e[0] + "*10^" + remove_e[1]
+        solved += "x^" + str(x - (count + 1)) + "*" + str(solution[count])
```
Comment (cclauss) [inline linear_algebra/src/polynom_for_points.py:94]:
```suggestion
        solved += f"x^{x - (count + 1)}*{solution[count]}"
```

File: linear_algebra/src/polynom_for_points.py [hunk 3]
```diff
@@ -24,96 +24,79 @@ def points_to_polynomial(coordinates: list[list[int]]) -> str:
...
-                solved += "+"
-            count += 1
+    count = 0
+    solved = "f(x)="
 
-        return solved
+    while count < x:
+        remove_e: list[str] = solution[count].split("E")
+        if len(remove_e) > 1:
+            solution[count] = remove_e[0] + "*10^" + remove_e[1]
```
Comment (cclauss) [inline linear_algebra/src/polynom_for_points.py:93]:
```suggestion
            solution[count] = f"{remove_e[0]}*10^{remove_e[1]}"
```

**Review by cclauss — CHANGES_REQUESTED** [review]
Nice!  https://www.scivision.dev/python-f-string-speed

**Review by cclauss — APPROVED** [review]
Awesome work!  Other pull requests that further reduce our `max-complexity` would be warmly received.

**Comment by will472** [comment]
Unsubscribe

**Comment by cclauss** [comment]
@will472 Please click the `unsubscribe` at the bottom of the right column of this page.

**Pull request merged**